modules/aws/vault/vault-iam.tf

```diff
@@ -1,6 +1,7 @@
 resource "aws_iam_role" "vault" {
-  name = "${var.cluster_name}-vault"
-  path = "/"
+  count = "${var.vault_auto_unseal ? 1 : 0}"
```

Comment:
we do have `IAM` in aws china, why do you want to disable it?

Reply:
Here the reason why I disable this for China. 

I've seen in the commit that these TF resources have been during the KMS Auto Unseal feature. 

Hence I thought that in case auto unseal is disabled like it must be in China these resources are not necessary and hence can be disabled.

If it is not the case then please can you provide here which is the case to have IAM Role when Auto Unseal is disabled?

Reply:
Yes, those resources are not required. But all of them were added non because of autounseal, but because cloudconfig size was increased (auto-unesal vault config part was added). And that required using same approach for user-data of the vm as with other nodes. So I think it is good to keep this aligned even if user-data is not going beyond size limit. Otherwise we will need to bring those resources back in case we added any other small piece of configuration for vault nodelater

Reply:
Now it makes more sense. 👍 for the insights. Then I made again the change